### PR TITLE
Doctests for Options methods

### PR DESCRIPTION
**Pull request by kaedroho**
Part of #69, but also applies to existing options methods

### CODE REVIEW

**Comment by kaedroho** [comment]
Looks like ``set_wal_recovery_mode`` doesn't work.

File: src/rocksdb_options.rs [hunk 1]
```diff
@@ -578,6 +818,15 @@ impl Options {
...
     /// Default: DBRecoveryMode::PointInTime
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use rocksdb::{Options. DBRecoveryMode};
+    ///
+    /// let mut opts = Options::default();
+    /// opts.set_wal_recovery_mode(DBRecoveryMode::AbsoluteConsistency);
+    /// ```
```
Comment (spacejam) [inline src/rocksdb_options.rs]: any ideas why this test is failing?

File: src/rocksdb_options.rs [hunk 2]
```diff
@@ -565,6 +796,15 @@ impl Options {
...
     /// Default: false
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use rocksdb::Options;
+    ///
+    /// let mut opts = Options::default();
+    /// opts.set_report_bg_io_stats(true);
+    /// ```
```
Comment (spacejam) [inline src/rocksdb_options.rs]: any idea why this fails on the 4.5 & 4.1 tests?

Comment (kaedroho) [inline src/rocksdb_options.rs]: Looks like this function was implemented recently. I can't find it in the 4.5 version of the CAPI: https://github.com/facebook/rocksdb/blob/4.5.fb/db/c.cc

Waiting for static linking to be merged would be the easiest solution to this.

File: src/rocksdb_options.rs (hunk 1 above)
Comment (kaedroho) [inline src/rocksdb_options.rs]: Just turned out to be a typo in the use statement. Fixed now

**Comment by kaedroho** [comment]
Rebased

**Comment by kaedroho** [comment]
``set_wal_recovery_mode`` seems to be broken in some other way on older versions of RocksDB. But everything works on 4.13 so we should be able to merge this after static linking is in.

**Comment by spacejam** [comment]
tests failing on 
```
test rocksdb_options::Options::compression_per_level_0 ... FAILED
test rocksdb_options::Options::set_compression_type_0 ... FAILED
test rocksdb_options::Options::set_wal_recovery_mode_0 ... FAILED
```

**Comment by kaedroho** [comment]
They should be fixed when I rebase this (will quickly do that now)

**Comment by spacejam** [comment]
the same 3 appear to be failing after rebase

**Comment by kaedroho** [comment]
Looks like the errors have become import errors now. I accidentally missed out a couple of ``pub use``'s in https://github.com/spacejam/rust-rocksdb/commit/e34692948eb786f9f4d93424e3f7ff482b82edf3 (they were merged into master after I did the initial work on static linking).

I'll reinstate them in this PR

**Comment by spacejam** [comment]
Cool! I'll merge as soon as tests pass!

**Comment by spacejam** [comment]
Thanks!

**Pull request merged**